provider/openstack/openstack_internal_test.go

```diff
@@ -0,0 +1,70 @@
+package openstack
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/flavors"
+
+	. "gopkg.in/check.v1"
+)
+
+func Test(t *testing.T) { TestingT(t) }
+
+type TestingOpenstackSuite struct{}
+
+var _ = Suite(&TestingOpenstackSuite{})
+
+func (s *TestingOpenstackSuite) TestPickOptimalFlavor(c *C) {
+	m := []struct {
+		comment string
+		flavors []flavors.Flavor
+		optimal string
+		err     string
+	}{
+		{
+			"no matching flavor",
+			[]flavors.Flavor{}, "", "No matching flavors to pick from",
+		},
+		{
+			"prefer smaller disk for equal memory",
+			[]flavors.Flavor{
+				flavors.Flavor{Disk: 30, RAM: 1, ID: "f01"},
+				flavors.Flavor{Disk: 10, RAM: 1, ID: "f02"},
+				flavors.Flavor{Disk: 20, RAM: 1, ID: "f03"},
+				flavors.Flavor{Disk: 40, RAM: 1, ID: "f04"},
+			}, "f02", "",
+		},
+		{
+			"prefer smaller disk regardless memory",
+			[]flavors.Flavor{
+				flavors.Flavor{Disk: 50, RAM: 1, ID: "f01"},
+				flavors.Flavor{Disk: 1, RAM: 16, ID: "f02"},
+				flavors.Flavor{Disk: 40, RAM: 1, ID: "f03"},
+				flavors.Flavor{Disk: 20, RAM: 1, ID: "f04"},
+			}, "f02", "",
+		},
+		{
+			"prefer smaller memory for equal hdd",
+			[]flavors.Flavor{
+				flavors.Flavor{Disk: 10, RAM: 3, ID: "f01"},
+				flavors.Flavor{Disk: 10, RAM: 2, ID: "f02"},
+				flavors.Flavor{Disk: 10, RAM: 1, ID: "f03"},
+				flavors.Flavor{Disk: 10, RAM: 8, ID: "f04"},
+			}, "f03", "",
+		},
+	}
+	for _, args := range m {
+		// Inform user about current case (will be hidden unless test fails).
+		fmt.Println(args.comment)
```

Comment:
Please use [`c.Log`](https://godoc.org/gopkg.in/check.v1#C.Log) instead of `Println`. The former will only print information when the test fails as in other cases you don't actually need this information.